distributed/core.py

```diff
@@ -316,7 +342,39 @@ def __init__(
         connection_args=None,
         timeout=None,
         io_loop=None,
+        local_directory=None,
     ):
+        from distributed.nanny import Nanny
+
+        if local_directory is None:
+            local_directory = (
+                dask.config.get("temporary-directory") or tempfile.gettempdir()
+            )
+
+        if "dask-worker-space" not in str(local_directory):
+            local_directory = os.path.join(local_directory, "dask-worker-space")
+
+        self._original_local_dir = local_directory
+
+        with warn_on_duration(
+            "1s",
+            "Creating scratch directories is taking a surprisingly long time. ({duration:.2f}s) "
+            "This is often due to running workers on a network file system. "
+            "Consider specifying a local-directory to point workers to write "
+            "scratch data to a local disk.",
+        ):
+            self._workspace = WorkSpace(local_directory)
+
+            if isinstance(self, Nanny):
+                self._workdir = None
+                self.local_directory = self._workspace.base_dir
+            else:
+                self._workdir = self._workspace.new_work_dir(prefix="worker-")
```

Comment:
"worker-" might be confusing as this applies to the scheduler now. Do you have thoughts on a possibly better name?

Reply:
Super, after adjusting the explicit check for Nanny, I think it plays pretty well taking the child's class name? https://github.com/dask/distributed/pull/7802/commits/830f4e0ad96d481c85033e67148d82f692673302